### PR TITLE
feat(compute): add compute disk create with snapshot schedule

### PR DESCRIPTION
**Pull request by TetyanaYahodska**
## Description
Documentation - https://cloud.google.com/compute/docs/disks/scheduled-snapshots?authuser=0#create_disk_add_policy
Fixes #<ISSUE-NUMBER>

Note: Before submitting a pull request, please open an issue for discussion if you are not associated with Google.

## Checklist

- [x] I have followed [Sample Format Guide](https://github.com/GoogleCloudPlatform/java-docs-samples/blob/main/SAMPLE_FORMAT.md)
- [ ] `pom.xml` parent set to latest `shared-configuration`
- [ ] Appropriate changes to README are included in PR
- [ ] These samples need a new **API enabled** in testing projects to pass (let us know which ones)
- [ ] These samples need a new/updated **env vars** in testing projects set to pass (let us know which ones)
- [x] **Tests** pass:   `mvn clean verify` **required**
- [x] **Lint**  passes: `mvn -P lint checkstyle:check` **required**
- [ ] **Static Analysis**:  `mvn -P lint clean compile pmd:cpd-check spotbugs:check` **advisory only**
- [ ] This sample adds a new sample directory, and I updated the [CODEOWNERS file](https://github.com/GoogleCloudPlatform/java-docs-samples/blob/main/.github/CODEOWNERS) with the codeowners for this sample
- [ ] This sample adds a new **Product API**, and I updated the [Blunderbuss issue/PR auto-assigner](https://github.com/GoogleCloudPlatform/java-docs-samples/blob/main/.github/blunderbuss.yml) with the codeowners for this sample 
- [ ] Please **merge** this PR for me once it is approved

### CODE REVIEW

File: compute/cloud-client/src/main/java/compute/disks/CreateDiskWithSnapshotSchedule.java [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+    String snapshotScheduleName = "YOUR_SCHEDULE_NAME";
+
+    createDiskWithSnapshotSchedule(projectId, zone, diskName, snapshotScheduleName);
+  }
+
+  // Creates disk with linked snapshot schedule.
+  public static Status createDiskWithSnapshotSchedule(
+      String projectId, String zone, String diskName, String snapshotScheduleName)
+      throws IOException, ExecutionException, InterruptedException, TimeoutException {
+    try (DisksClient disksClient = DisksClient.create()) {
```
Comment (Sita04) [inline compute/cloud-client/src/main/java/compute/disks/CreateDiskWithSnapshotSchedule.java:50]: Add the client initializer comment here.

File: compute/cloud-client/src/main/java/compute/disks/CreateDiskWithSnapshotSchedule.java [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+
+      // Get the resource policy to link to the disk
+      String resourcePolicyLink = String.format("projects/%s/regions/%s/resourcePolicies/%s",
+              projectId, zone.substring(0, zone.lastIndexOf('-')), snapshotScheduleName);
```
Comment (Sita04) [inline compute/cloud-client/src/main/java/compute/disks/CreateDiskWithSnapshotSchedule.java:54]: Please get the snapshot schedule region as a separate parameter. The snapshot schedule doesn't have to be in the same region as the disk, iiuc?

File: compute/cloud-client/src/main/java/compute/snapshotschedule/CreateSnapshotSchedule.java [hunk 3]
```diff
@@ -0,0 +1,134 @@
...
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package compute.snapshotschedule;
+
+// [START compute_snapshot_schedule_create]
```
Comment (Sita04) [inline compute/cloud-client/src/main/java/compute/snapshotschedule/CreateSnapshotSchedule.java:19]: Why is this sample duplicated?
It is already present in: https://github.com/GoogleCloudPlatform/java-docs-samples/pull/9742

File: compute/cloud-client/src/main/java/compute/snapshotschedule/DeleteSnapshotSchedule.java [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package compute.snapshotschedule;
+
+// [START compute_snapshot_schedule_delete]
```
Comment (Sita04) [inline compute/cloud-client/src/main/java/compute/snapshotschedule/DeleteSnapshotSchedule.java:19]: Why is this sample duplicated?
It is already present in: https://github.com/GoogleCloudPlatform/java-docs-samples/pull/9742

File: compute/cloud-client/src/test/java/compute/disks/DisksIT.java [hunk 5]
```diff
@@ -305,9 +317,19 @@ public void testDiskAttachResize()
...
   }
+
+  @Test
+  void testCreateDiskWithSnapshotSchedule()
+          throws IOException, ExecutionException, InterruptedException, TimeoutException {
+    Status status = CreateDiskWithSnapshotSchedule.createDiskWithSnapshotSchedule(
+            PROJECT_ID, ZONE, DISK_WITH_SNAPSHOT_SCHEDULE, SNAPSHOT_SCHEDULE);
+
+    Assert.assertNotNull(Util.getDisk(PROJECT_ID, ZONE, DISK_WITH_SNAPSHOT_SCHEDULE));
+    assertThat(status).isEqualTo(Operation.Status.DONE);
```
Comment (Sita04) [inline compute/cloud-client/src/test/java/compute/disks/DisksIT.java:333]: This assertion is not necessary.

**Review by Sita04 — CHANGES_REQUESTED** [review]

File: compute/cloud-client/src/main/java/compute/disks/CreateDiskWithSnapshotSchedule.java (hunk 1 above)
Comment (TetyanaYahodska) [reply compute/cloud-client/src/main/java/compute/disks/CreateDiskWithSnapshotSchedule.java:50]: Thank you! Added comment

File: compute/cloud-client/src/main/java/compute/disks/CreateDiskWithSnapshotSchedule.java (hunk 2 above)
Comment (TetyanaYahodska) [reply compute/cloud-client/src/main/java/compute/disks/CreateDiskWithSnapshotSchedule.java:54]: Thank you! Created region variable. 
ZONE: the location where you are creating the disk. The disk must be in a zone that is in the same region as the snapshot schedule.

File: compute/cloud-client/src/main/java/compute/snapshotschedule/CreateSnapshotSchedule.java (hunk 3 above)
Comment (TetyanaYahodska) [reply compute/cloud-client/src/main/java/compute/snapshotschedule/CreateSnapshotSchedule.java:19]: I need this classes for testing.

File: compute/cloud-client/src/main/java/compute/snapshotschedule/DeleteSnapshotSchedule.java (hunk 4 above)
Comment (TetyanaYahodska) [reply compute/cloud-client/src/main/java/compute/snapshotschedule/DeleteSnapshotSchedule.java:19]: I need this classes for testing.

File: compute/cloud-client/src/test/java/compute/disks/DisksIT.java (hunk 5 above)
Comment (TetyanaYahodska) [reply compute/cloud-client/src/test/java/compute/disks/DisksIT.java]: Deleted assertion

File: compute/cloud-client/src/test/java/compute/disks/DisksIT.java [hunk 6]
```diff
@@ -353,4 +363,15 @@ public void testCreateDiskSecondaryZonal()
...
+
+  @Test
+  void testCreateDiskWithSnapshotSchedule()
+          throws IOException, ExecutionException, InterruptedException, TimeoutException {
+    Status status = CreateDiskWithSnapshotSchedule.createDiskWithSnapshotSchedule(
+            PROJECT_ID, ZONE, DISK_WITH_SNAPSHOT_SCHEDULE, SNAPSHOT_SCHEDULE);
+    Disk disk = Util.getDisk(PROJECT_ID, ZONE, DISK_WITH_SNAPSHOT_SCHEDULE);
+
+    assertThat(status).isEqualTo(Status.DONE);
+    assertEquals(DISK_WITH_SNAPSHOT_SCHEDULE, disk.getName());
```
Comment (Sita04) [inline compute/cloud-client/src/test/java/compute/disks/DisksIT.java:375]: This assertion basically checks that the getDisk returns the correct disk. It does not verify if the sample.

I'd replace this assertion to verify that the Util.getDisk doesn't throw an exception when retrieving the disk.

**Review by Sita04 — CHANGES_REQUESTED** [review]

File: compute/cloud-client/src/test/java/compute/disks/DisksIT.java (hunk 6 above)
Comment (TetyanaYahodska) [reply compute/cloud-client/src/test/java/compute/disks/DisksIT.java:375]: Thank you! Fixed

File: compute/cloud-client/src/main/java/compute/snapshotschedule/CreateSnapshotSchedule.java (hunk 3 above)
Comment (TetyanaYahodska) [reply compute/cloud-client/src/main/java/compute/snapshotschedule/CreateSnapshotSchedule.java]: Thank you! Deleted

File: compute/cloud-client/src/main/java/compute/snapshotschedule/DeleteSnapshotSchedule.java (hunk 4 above)
Comment (TetyanaYahodska) [reply compute/cloud-client/src/main/java/compute/snapshotschedule/DeleteSnapshotSchedule.java]: Thank you! Deleted

**Review by Sita04 — APPROVED** [review]

**Pull request merged**